src/client/websocket/handlers/TYPING_START.js

```diff
@@ -1,15 +1,15 @@
 'use strict';
-
 const { Events } = require('../../../util/Constants');
 
 module.exports = (client, { d: data }) => {
   const channel = client.channels.cache.get(data.channel_id);
   const user = client.users.cache.get(data.user_id);
   const timestamp = new Date(data.timestamp * 1000);
+  const textBasedChannelTypes = ['dm', 'text', 'news'];
 
   if (channel && user) {
-    if (channel.type === 'voice') {
-      client.emit(Events.WARN, `Discord sent a typing packet to a voice channel ${channel.id}`);
+    if (!textBasedChannelTypes.includes(channel.type)) {
+      client.emit(Events.WARN, `Discord sent a typing packet to a ${channel.type} channel ${channel.id}`);
```

Comment:
```suggestion
      client.emit(Events.WARN, `Discord sent a typing packet to a ${channel.type} channel: ${channel.id}`);
```

Reply:
I'll gladly accept the formatting here as i originally planned on applying it, if i can get a confirmation that the warn event value is not part of semver classifications, as technically the `:` means a change in public facing behavior which (very very technically) is a breaking change upping this to major

Reply:
after internal discussion:
since one might actually rely on the return values from the "warn" event it is considered a public facing API making any breaking change a semver major.

Thanks for the suggestion, yes it looks nicer, but won't be applied here

--- 

Since i keep receiving questions about it (and why i make such a big fuss about a ":") on the server:
With an existing voice channel the WARN value would change in this way:

```diff
- Discord sent a typing packet to a voice channel 222078108977594369
+ Discord sent a typing packet to a voice channel: 222078108977594369
```

The ":" was not there and might (granted, very very unlikely) change (break) the existing behavior for someone. Consider this regular expression, for a very crude example: 

```js
/Discord sent a typing packet to a voice channel (\d+)/
```
[[link to regex101 with the example]](https://regex101.com/r/XBxG5p/1)

It would not trigger after applying this change. Breaking the functionality for the person using it.

Without the addition of the ":" the behavior for a given voice channel will be the same. However it will also allow non-voice channels that do not have typing capability to be filtered, fixing the bug causing the error the user has experienced and reported (see [OP](https://github.com/discordjs/discord.js/pull/4349#issue-432424060))

As per https://semver.org/ 
> 1. MAJOR version when you make incompatible API changes,

So, yes, even this little additional ":" introduces an incompatible change of behavior of a public facing API, which would up this to a semver major (meaning the earliest release being 13.0.0). Since this is as a bug fix however way too important to hold back that long this change will not be applied.

The rest of this PR will then be compatible (see reasoning above), keeping this at semver patch

> 3. PATCH version when you make backwards compatible bug fixes.

Meaning it can be released as early as `12.2.x` instead of `13.0.0`